app/main/views/root.py

```diff
@@ -77,15 +77,13 @@ def redirect_to_location(current_location):
     """
 
     if current_location == 'introduction':
-        return redirect(url_for("main.landing_page"))
-    elif current_location == "completed":
-        return redirect(url_for("main.submission"))
-    elif current_location == "submitted":
-        return redirect(url_for("main.thank_you"))
-    elif current_location == 'questionnaire':
-        return redirect(url_for("main.questionnaire"))
+        return redirect('/questionnaire/introduction')
+    elif current_location == "summary":
+        return redirect('/questionnaire/summary')
+    elif current_location == "thank-you":
+        return redirect('/questionnaire/thank-you')
     else:
-        return errors.page_not_found()
+        return redirect('/questionnaire/' + current_location)
```

Comment:
Yeah agreed this block looks a little strange. 

However I think we can remove the entire method now, it's only referenced by the login method. That can just do a redirect redirect('/questionnaire/' + current_location)